### PR TITLE
Add forwarding to Operator.sol

### PR DESCRIPTION
**Pull request by RyanRHall**
Addresses [#173542291](https://www.pivotaltracker.com/story/show/173542291)

This PR also refactors the `v0.7` contract directory layout to be consistent with the other versioned directories.

**Gas Benchmarking**

Tested against the `#setBytes` method on the `GetterSetter` contract.

```
#setBytes, bytes: 10, gasUsed: 45526
#foreward --> #setBytes, bytes: 10, gasUsed: 49404

#setBytes, bytes: 100, gasUsed: 128298
#foreward --> #setBytes, bytes: 100, gasUsed: 132194

#setBytes, bytes: 300, gasUsed: 253891
#foreward --> #setBytes, bytes: 300, gasUsed: 257835

#setBytes, bytes: 1000, gasUsed: 713923
#foreward --> #setBytes, bytes: 1000, gasUsed: 718001

#setBytes, bytes: 3000, gasUsed: 2010680
#foreward --> #setBytes, bytes: 3000, gasUsed: 2015147

#setBytes, bytes: 10000, gasUsed: 6591122
#foreward --> #setBytes, bytes: 10000, gasUsed: 6597084
```

**New Gas Benchmarking**
\* after making @kaleofduty's suggested solidity changes
```
#setBytes, bytes: 10, gasUsed: 45526
#forward --> #setBytes, bytes: 10, gasUsed: 48651

#setBytes, bytes: 100, gasUsed: 128298
#forward --> #setBytes, bytes: 100, gasUsed: 131441

#setBytes, bytes: 300, gasUsed: 253879
#forward --> #setBytes, bytes: 300, gasUsed: 257070

#setBytes, bytes: 1000, gasUsed: 713887
#forward --> #setBytes, bytes: 1000, gasUsed: 717212

#setBytes, bytes: 3000, gasUsed: 2010716
#forward --> #setBytes, bytes: 3000, gasUsed: 2014430

#setBytes, bytes: 10000, gasUsed: 6591134
#forward --> #setBytes, bytes: 10000, gasUsed: 6596343
```

Discrepancy is more noticeable for smaller payloads. Should be < 10% increase for most TXs.

### CODE REVIEW

File: evm-contracts/src/v0.7/dev/Operator.sol [hunk 1]
```diff
@@ -266,6 +266,15 @@ contract Operator is
     return address(s_LinkToken);
   }
 
+  function foreward(address _to, bytes calldata _data)
```
Comment (kaleofduty) [inline evm-contracts/src/v0.7/dev/Operator.sol:269]: "forward" ?

File: evm-contracts/src/v0.7/dev/Operator.sol [hunk 2]
```diff
@@ -266,6 +266,15 @@ contract Operator is
...
+    public
+    onlyAuthorizedNode()
+  {
+    require(_to != address(s_LinkToken), "Cannot use #foreward to send messages to Link token");
```
Comment (kaleofduty) [inline evm-contracts/src/v0.7/dev/Operator.sol:273]: gas optimization: make s_LinkToken `immutable`.

File: evm-contracts/src/v0.7/dev/Operator.sol [hunk 3]
```diff
@@ -266,6 +266,15 @@ contract Operator is
     return address(s_LinkToken);
   }
 
+  function foreward(address _to, bytes calldata _data)
+    public
+    onlyAuthorizedNode()
```
Comment (kaleofduty) [inline evm-contracts/src/v0.7/dev/Operator.sol:271]: `onlyAuthorizedNode` does two storage reads:

```
  modifier onlyAuthorizedNode() {
    require(s_authorizedNodes[msg.sender] || msg.sender == owner, "Not an authorized node to fulfill requests");
    _;
  }
```

We could save 800 gas on each call by getting rid of the `owner` part.

**Review by kaleofduty — COMMENTED** [review]
Some ideas for saving ~1'600 gas on each call to `forward`

File: evm-contracts/src/v0.7/dev/Operator.sol (hunk 3 above)
Comment (se3000) [reply evm-contracts/src/v0.7/dev/Operator.sol:271]: @kaleofduty Doesn't it short-circuit and avoid the second storage read? As the owner it seems helpful to allow the owner use the method as well. Alternatively, we could break out another forwarding method for the owner and avoid that first gas read.

File: evm-contracts/src/v0.7/dev/Operator.sol (hunk 2 above)
Comment (se3000) [reply evm-contracts/src/v0.7/dev/Operator.sol:273]: I'm in favor of making it immutable, but I think the convention would be to remove the `s_` prefix as well.

File: evm-contracts/src/v0.7/dev/Operator.sol (hunk 3 above)
Comment (kaleofduty) [reply evm-contracts/src/v0.7/dev/Operator.sol:271]: It does, not sure what was going on with my brain yesterday eve 🤦 
So never mind.

File: evm-contracts/src/v0.7/dev/Operator.sol (hunk 2 above)
Comment (kaleofduty) [reply evm-contracts/src/v0.7/dev/Operator.sol:273]: sgtm

File: evm-contracts/src/v0.7/dev/Operator.sol (hunk 1 above)
Comment (RyanRHall) [reply evm-contracts/src/v0.7/dev/Operator.sol:269]: 🤦‍♂️🤦‍♂️

File: evm-contracts/src/v0.7/dev/Operator.sol (hunk 3 above)
Comment (RyanRHall) [reply evm-contracts/src/v0.7/dev/Operator.sol:271]: oh I like that. And the owner just has to add themselves as an authorizedNode if they want to use the function?

Comment (RyanRHall) [reply evm-contracts/src/v0.7/dev/Operator.sol:271]: @kaleofduty - is the || lazy evaluated in solidity? Aka if the first part is true will solidity avoid the 2nd storage read?

**Review by se3000 — APPROVED** [review]

**Review by se3000 — APPROVED** [review]

**Pull request merged**